lib/curl_get_line.c

```diff
@@ -43,12 +43,22 @@ char *Curl_get_line(char *buf, int len, FILE *input)
     char *b = fgets(buf, len, input);
     if(b) {
       size_t rlen = strlen(b);
-      if(rlen && (b[rlen-1] == '\n')) {
+      if(rlen && ((b[rlen-1] == '\n') || feof(input))) {
         if(partial) {
           partial = FALSE;
           continue;
         }
-        return b;
+        if(b[rlen-1] != '\n') {
+          /* make sure line is '\n' terminated if EOF */
```

Comment:
This doesn't actually know that it is EOF, it could also be a line longer than 'len'. If you want it to act only on EOF, then the feof() check probably needs to set a separate flag for that purpose.

Reply:
I admit it's a bit hacky, but from man fgets, I get that the line will never be more than len chars long.

```
       char *fgets(char *s, int size, FILE *stream);

       fgets()  reads  in at most one less than size characters from stream and stores them into the buffer pointed to by s.  Reading stops after an EOF or a
       newline.  If a newline is read, it is stored into the buffer.  A terminating null byte ('\0') is stored after the last character in the buffer.
```

Reply:
> The man page explains "Reading stops after an EOF or a  newline."

If the line is `len` + newline or `len` + EOF we get the same `len` number of bytes without newline in the buffer. You then cannot know if the missing newline is because of too long buffer or if at EOF, just by checking for missing newline. If the line is *shorter* than `len` and is missing a newline, you know it is EOF.

Reply:
You're right, back to the drawing board.

Reply:
I now see it is impossible to know whether the last line is correct without parsing it when it completely fills the buffer, so I chose to opt for the most likely case: assume it is truncated.

As this makes it unreliable I could see why you would not want to allow a last non-newline line at all. 

Ignoring them does potentially break old config files though.

Reply:
@bagder Pushed a new commit, fixing the missing of truncated lines.